### DISCUSSION TITLE
Is the `gcm.message_id` check required to work?

### DISCUSSION THREAD

**Discussion by wmatyjanomtek** [invertase/react-native-firebase#6212] [Ideas]
Hello, 
In iOS AppDelegate.m in `didReceiveRemoteNotification` there is a code: 

```objective-c
  if (userInfo[@"gcm.message_id"]) {
    DLog(@"didReceiveRemoteNotification gcm.message_id was present %@", userInfo);
```

Is this code really needed? I don't think it's actually used anywhere, except from this Log. 
I'm using sendbird sdk which doesn't provide this field and I'm wondering whose issue it actually is.

**Comment by mikehardy** [comment]
:thinking: 

```
mike@bistromath:~/work/Invertase/react-native-firebase (main) % grep -r gcm.message_id

packages/messaging/ios/RNFBMessaging/RNFBMessagingSerializer.m:    if ([key isEqualToString:@"gcm.message_id"] || [key isEqualToString:@"google.message_id"] ||
packages/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m:  if (notification.request.content.userInfo[@"gcm.message_id"]) {
packages/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m:  if (remoteNotification[@"gcm.message_id"]) {
packages/messaging/ios/RNFBMessaging/RNFBMessaging+AppDelegate.m:  if (userInfo[@"gcm.message_id"]) {
packages/messaging/ios/RNFBMessaging/RNFBMessaging+AppDelegate.m:    DLog(@"didReceiveRemoteNotification gcm.message_id was present %@", userInfo);
```

I believe no attempt is made to process non-GCM messages, as we may not know how to deal with them. This may not be the most useful stance, I believe others have tried to integrate non-GCM (read as: non-firebase) messages and perhaps something could be done to be a bit more open.

However, it may be better to handle your non-firebase notification needs via Notifee.app ? I don't *think* it restricts on gcm.message_id

Not sure how useful this comment is, as I'm not sure of your use case

**Reply by wmatyjanomtek** [reply]
Hello, @mikehardy ,
Thanks for your reply.
These actually are firebase notifications (at least they were supposed to be like this :P)
Look: they actually ARE linking to the rn firebase docs in their dosc: https://sendbird.com/docs/chat/v3/javascript/guides/push-notifications

And I wonder: whose fault is this, that it doesn't work as expected? I suppose it's on their side.

**Reply by mikehardy** [reply]
Wow, if they're recommending both react-native-firebase and notifee (both Invertase projects, I work on both) I'm going to guess they've run through their sample code and tested the integration and it works.

As such, I would expect that they are actually posting to FCM via the FCM REST API and it will have the message_id, or it should because that's from the Google Firebase servers

I think there is something else going on, but of course with proprietary products (sendbird, the firebase servers themselves, the APNs middle-tier that Apple routes through) I just can't be sure.

But just from first assumptions: if sendbird is recommending these packages, it seems they should be working and they should know how to make it work. I'd work with them
